src/stores/right-panel/RightPanelStore.ts

```diff
@@ -230,11 +230,19 @@ export default class RightPanelStore extends ReadyWatchingStore {
     }
 
     private emitAndUpdateSettings() {
+        if (this.global?.history) {
```

Comment:
perhaps this filter stage can be a private method you call e.g `this.filterValidHistory()` or similar as the same if/filter is duplicated